crypto/x509v3/v3_akey.c

```diff
@@ -42,7 +42,7 @@ static STACK_OF(CONF_VALUE) *i2v_AUTHORITY_KEYID(X509V3_EXT_METHOD *method,
     char *tmp;
     if (akeyid->keyid) {
         tmp = OPENSSL_buf2hexstr(akeyid->keyid->data, akeyid->keyid->length);
-        X509V3_add_value("keyid", tmp, &extlist);
+        X509V3_add_value(akeyid->issuer || akeyid->serial ? "keyid" : NULL, tmp, &extlist);
```

Comment:
What does `X509V3_add_value` do when the first arg is NULL?

Reply:
Pleased that you had a look at this PR.

Would be good if functions like `X509V3_add_value`, which appear in the public headers, were documented ;-)

I has been a while that I've been doing this patch, but I'm pretty sure that a `NULL` pointer is fine at this place, and then simply no name is printed, for several reasons:
* This function does get called with `NULL` `name` arg also elsewhere, for instance:
```
crypto/x509v3/v3_tlsf.c:            X509V3_add_value(NULL, tls_feature_tbl[j].name, &ext_list);
crypto/x509v3/v3_tlsf.c:            X509V3_add_value_int(NULL, ai, &ext_list);
```
* The function checks the `name` parameter for `NULL` in way that suggests that this parameter is optional.
* As far as I recall I tried out also the `NULL` case.

Reply:
BTW, the idea for calling `X509V3_add_value()` at the above quoted place with an `NULL` pointer in case neither `keyid->issuer` nor `akeyid->serial` are present was that in this case it should be clear that the hex value printed is just a simple authority key identifier, and then the value can be printed in the very same way as for the subject key identifier. In the given example, 
```
            X509v3 Subject Key Identifier:
                11:49:46:19:2A:4E:4D:D1:C8:FB:79:55:3D:81:99:22:EE:34:4F:22
            X509v3 Authority Key Identifier:
                keyid:11:49:46:19:2A:4E:4D:D1:C8:FB:79:55:3D:81:99:22:EE:34:4F:22
```
becomes simpler (and to most user less confusing) to read:
```
            X509v3 Subject Key Identifier:
                11:49:46:19:2A:4E:4D:D1:C8:FB:79:55:3D:81:99:22:EE:34:4F:22
            X509v3 Authority Key Identifier:
                11:49:46:19:2A:4E:4D:D1:C8:FB:79:55:3D:81:99:22:EE:34:4F:22
```

Reply:
Could we add some parens in here (prior to ?).
I actually find it a bit weird that we would sometimes print keyid and not other times. Would it be better for them both to always print keyid, or (alternatively) for neither of them to ever do it.

Reply:
Sure we could place parentheses around `keyid->issuer || akeyid->serial` 
but according to your comment I meanwhile also see that for uniformity the `keyid:` tag should better be generally left out (which for instance is done also by Mozilla's cert viewer). 
Since moreover the interpretation of the raw ID data should be obvious from the line preceding it (`X509v3 Authority Key Identifier:`) I've just removed the `keyid:` tag unconditionally.